### PR TITLE
ci: version to 3.0.0 Micronaut 4.0.0-SNAPSHOT githubCoreBranch=4.0.x

### PR DESCRIPTION
**Pull request by wetted**

### CODE REVIEW

File: .github/workflows/graalvm.yml [hunk 1]
```diff
@@ -20,6 +20,7 @@ jobs:
     strategy:
       matrix:
         java: ['11', '17']
+        graalvm: ['latest', 'dev']
```
Comment (sdelamo) [inline .github/workflows/graalvm.yml:23]: The changes to .graalvm should be in PR against master

File: settings.gradle [hunk 2]
```diff
@@ -46,10 +46,10 @@ enableFeaturePreview 'TYPESAFE_PROJECT_ACCESSORS'
 dependencyResolutionManagement {
     repositories {
         mavenCentral()
+        maven { url "https://s01.oss.sonatype.org/content/repositories/snapshots/" }
     }
-    versionCatalogs {
```
Comment (sdelamo) [inline settings.gradle:50]: can you submit these changes to `master` as well.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: settings.gradle (hunk 2 above)
Comment (wetted) [reply settings.gradle:50]: https://github.com/micronaut-projects/micronaut-oracle-cloud/pull/412

**Comment by wetted** [comment]
I had to make some changes to the following classes, due to changed APIs in core and deprecated API that was removed. The following specifically need to be addressed further. I wasn't entirely sure what to do with them.

`io.micronaut.oraclecloud.function.http.FnServletRequest`
`io.micronaut.oraclecloud.function.http.FnServletResponse.FnResponseHeaders.setConversionService`

**Pull request merged**